packages/design-system/src/components/text-field.tsx

```diff
@@ -314,8 +328,8 @@ export const TextField = React.forwardRef<HTMLDivElement, TextFieldProps>(
           ref={mergeRefs(internalInputRef, inputRef ?? null)}
         />
 
-        {prefix && <PrefixSlot>{prefix}</PrefixSlot>}
-        {suffix && <SuffixSlot>{suffix}</SuffixSlot>}
+        {prefix && <PrefixSlot ref={prefixRef}>{prefix}</PrefixSlot>}
```

Comment:
you control the prefix element outside, so you should be getting the ref to it instead of using the slot